src/Libraries/Samples/SampleIntegration/TracedHog.cs

```diff
@@ -92,13 +94,20 @@ private TracedHog(double x, double y, int id)
         this.ID = id;
 
         TracedHogManager.RegisterHogForID(id, this);
+
+        WorkspaceEvents.WorkspaceAdded += WorkspaceEventsWorkspaceAdded;
+    }
+
+    void WorkspaceEventsWorkspaceAdded(WorkspacesModificationEventArgs args)
```

Comment:
Could you add a section to the Host integration guide on the wiki of "events and when to use them"? If you don't have the cycles, let me know and I'll try and do it.